core/domain/story_services.py

```diff
@@ -535,3 +536,81 @@ def save_story_summary(story_summary):
     )
 
     story_summary_model.put()
+
+
+def get_completed_node_ids(user_id, story_id):
+    """Returns the ids of the nodes completed in the story.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_id: str. ID of the story.
+
+    Returns:
+        list(str). List of the node ids completed in story.
+    """
+    progress_model = user_models.StoryProgressModel.get(user_id, story_id)
+
+    return progress_model.completed_nodes if progress_model else []
+
+
+def get_node_ids_completed_in_stories(user_id, story_ids):
+    """Returns the ids of the nodes completed in each of the stories.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_ids: list(str). IDs of the stories.
+
+    Returns:
+        dict(list(str)). Dict of the story id and the node ids completed
+            in each story as key-value pairs.
+    """
+    progress_models = user_models.StoryProgressModel.get_multi(
+        user_id, story_ids)
+
+    node_ids_completed_in_stories = {}
+
+    for progress_model in progress_models:
+        if progress_model:
+            node_ids_completed_in_stories[progress_model.story_id] = (
+                progress_model.completed_nodes)
```

Comment:
If completed_nodes contains node IDs then we should call it completed_node_ids instead.

Reply:
changed